tb/cli.py

```diff
@@ -18,28 +20,45 @@
 __credits__ = "adjust OS team"
 
 
-import logging
-import pydoc
-import sys
+class Tinderbox:
+    """tinderbox: a Python script to drive the binary package compilation process.
 
-import docopt
+Usage:
+  tinderbox [-h|--help] [-v|--version] <command>
 
-import tb.backend as tbb
+Commands:
+  tinderbox clean_targets   clean targets directory.
+  tinderbox create_targets  create targets directory.
+  tinderbox prepare         compile base packages for all targets.
+  tinderbox compile         compile system and world packages for all targets.
+  tinderbox all             all of the above.
 
-logging.basicConfig(
-    level=logging.INFO, format="[%(asctime)s][%(name)s][%(levelname)s]: %(message)s"
-)
-logger = logging.getLogger(__name__)
+Options:
+    -h,  --help        display help.
+    -v,  --version     display version."""
+
+    def __init__(self):
+        args = docopt.docopt(
+            Tinderbox.__doc__, version=__version__, help=True, options_first=True
+        )
 
+        runner = {
+            "clean_targets": tbb.clean_targets,
+            "create_targets": tbb.create_targets,
+            "prepare": tbb.prepare,
+            "compile": tbb.compile_packages,
+            "all": tbb.run,
+        }
 
-def main():
-    """Main function."""
-    argv = docopt.docopt(__doc__)
-    logger.info("Running tinderbox with the following options:")
-    logger.info(dict(argv))
-    tbb.run()
-    return 0
+        try:
+            runner[args["<command>"]]()
+        except KeyError:
+            sys.exit(
+                """Error! \'%s\' is an invalid command!
+See 'tinderbox --help' for a complete list of commands."""
+                % (args["<command>"])
+            )
```

Comment:
I think we could simplify this a bit, how about this?
```suggestion
        cmd = args["<command>"]
        if cmd not in runner:
            sys.exit(f"""Error! '{cmd}' is an invalid command!
See 'tinderbox --help' for a complete list of commands.""")
        runner[cmd]()
```

Reply:
I prefer the try catch version 👀